src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala

```diff
@@ -55,7 +53,51 @@ class AgoraMongoDao(collections: Seq[MongoCollection]) extends AgoraDao {
     //insert the entity
     val dbEntityToInsert = EntityToMongoDbObject(entityWithId)
     collection.insert(dbEntityToInsert)
-    findSingle(entityWithId)
+    val insertedEntity = findSingle(MongoDbObjectToEntity(dbEntityToInsert))
+    insertedEntity
+  }
+
+  override def findSingle(namespace: String, name: String, snapshotId: Int): AgoraEntity = {
```

Comment:
I reordered some methods, so you'll see additions and deletions due purely to those moves. (I moved all the interface methods to the top).